### ISSUE TITLE
Python exception when trying to login : "(insecure_transport) OAuth 2 MUST utilize https"

### ISSUE THREAD

**Issue by TijsBusschaert** [tillsteinbach/WeConnect-python#215]
This fault started to occur recently without any clear cause.
The car is an VW ID.4 from 2023
The VW connect app still works as before.

**Libraries installed**
certifi            2024.8.30
charset-normalizer 3.4.0
idna               3.10
oauthlib           3.2.2
pip                24.3.1
requests           2.32.3
RPi.GPIO           0.7.1
setuptools         66.1.1
urllib3            2.2.3
weconnect          0.60.5

Any ideas on how to troubleshoot or solve?

**Comment by tillsteinbach** [comment]
Is this still the case after a restart?

**Comment by Lake292** [comment]
Same behavior after restart for me. Tried manually running all vehicles example or even through home assistant.

**Comment by DiskoPete** [comment]
Hi

I noticed that we don't get a full URL here anymore but an absolute path `/signin-service/v1/...`
https://github.com/tillsteinbach/WeConnect-python/blob/main/weconnect/auth/we_connect_session.py#L247

This will fail the 'is_secure_transport' check here
https://github.com/oauthlib/oauthlib/blob/master/oauthlib/oauth2/rfc6749/utils.py#L79-L83

Anyways, the 'terms-and-conditions' segment in the path made me wonder if there is some new agreement.

I logged in to https://myvolkswagen.net/ via browser and there I had to accept some agreement.
Now the error is gone.

This is the full path the request returned
`/signin-service/v1/{...}@apps_vw-dilab_com/terms-and-conditions?relayState={...}&canChangeCountryOfResidence=false&userId={...}&updated=dataprivacy`

(I replaced sensitive data with `{...}`

Maybe we can adapt the error message to give a hint that a new agreement must be accepted? 🤷

**Comment by TijsBusschaert** [comment]
When did you login to https://myvolkswagen.net/ ? 
I cannot access this page, I get error 404.
Seems like VW is stirring some things up.

**Comment by kipe** [comment]
I can confirm that the same solution worked for me.
https://myvolkswagen.net redirected me to the Finnish page and once I logged in and accepted the terms restarting the container was enough and everything seems to be working once again.

It's quite weird that the mobile software once again kept working without accepting new terms :S

**Comment by DiskoPete** [comment]
> It's quite weird that the mobile software once again kept working without accepting new terms :S

@kipe I agree. There must be some way to skip the agreements, since the mobile app still can connect.

Unfortunately, I have no clue how to debug this. 😄

**Comment by tillsteinbach** [comment]
You will probably also see this when logging out and in again in the app.
